distribution/pull_v2.go

```diff
@@ -658,18 +658,21 @@ func (p *v2Puller) pullManifestList(ctx context.Context, ref reference.Named, mf
 		return "", "", err
 	}
 
+	logrus.Debugf("%s resolved to a manifestList object with %d entries; looking for a os/arch match", ref, len(mfstList.Manifests))
 	var manifestDigest digest.Digest
 	for _, manifestDescriptor := range mfstList.Manifests {
 		// TODO(aaronl): The manifest list spec supports optional
 		// "features" and "variant" fields. These are not yet used.
 		// Once they are, their values should be interpreted here.
 		if manifestDescriptor.Platform.Architecture == runtime.GOARCH && manifestDescriptor.Platform.OS == runtime.GOOS {
 			manifestDigest = manifestDescriptor.Digest
+			logrus.Debugf("found match; digest %s for %s/%s", manifestDigest.String(), runtime.GOOS, runtime.GOARCH)
 			break
 		}
 	}
 
 	if manifestDigest == "" {
+		logrus.Debugf("unable to find match for %s/%s in manifest list entries", runtime.GOOS, runtime.GOARCH)
```

Comment:
good idea @cpuguy83 ..let me get this PR updated with the other comment as well!